packages/core/database/lib/index.js

```diff
@@ -58,9 +58,17 @@ class Database {
     return this.entityManager.getRepository(uid);
   }
 
-  getConnection(tableName) {
+  async transaction(cb) {
+    await this.connection.transaction(async trx => {
+      const em = createEntityManager(this, trx);
+      await cb(em);
+    });
+  }
+
+  getConnection(tableName, trx) {
     const schema = this.connection.getSchemaName();
-    const connection = tableName ? this.connection(tableName) : this.connection;
+    const connOrTrx = trx ? trx : this.connection;
+    const connection = tableName ? connOrTrx(tableName) : connOrTrx;
     return schema ? connection.withSchema(schema) : connection;
```

Comment:
The `getConnection` method should take the lock option and return locked transacting connection.

```diff
-getConnection(tableName, trx) {
+getConnection(tableName, trx, lock) {
    const schema = this.connection.getSchemaName();
    const connOrTrx = trx ? trx : this.connection;
    const connection = tableName ? connOrTrx(tableName) : connOrTrx;
-    return schema ? connection.withSchema(schema) : connection;
+    const schemaConn = schema ? connection.withSchema(schema) : connection;
+    if (lock) {
+      return lock === 'exclusive' ? schemaConn.transacting(trx).forUpdate() : schemaConn.transacting(trx).forShare()
+    }
+    return schemaConn;
}
```